### PR TITLE
Remove transaction surrounding `createAPIKey` call

### PR DESCRIPTION
**Pull request by tempoz**
**Related issues**: N/A

### CODE REVIEW

**Comment by tempoz** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/6221?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6221** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/6221?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6192** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/6192?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6180** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/6180?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6045** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/6045?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6044** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/6044?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6043** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/6043?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6014** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/6014?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6012** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/6012?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6010** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/6010?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6003** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/6003?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5865** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/5865?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#5875](https://github.com/buildbuddy-io/buildbuddy/pull/5875) <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/5875?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#6002** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/6002?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tempoz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: enterprise/server/secrets/secrets.go [hunk 1]
```diff
@@ -113,6 +113,10 @@ func (s *SecretService) ListSecrets(ctx context.Context, req *skpb.ListSecretsRe
 	return rsp, nil
 }
 
+// UpdateSecret updates the secret with the given name associated with the group
+// the user is authenticated as, creating it if it does not exist. It returns,
+// in order, the response proto, whether the secret was newly created or not,
+// and any error encountered.
```
Comment (jdhollen) [inline enterprise/server/secrets/secrets.go:119]: unrelated?

Comment (tempoz) [reply enterprise/server/secrets/secrets.go]: mmm, good call. I must have added this comment while reading code and then forgotten to make its own doc PR. i'll take it out.
